journal/method.hpp

```diff
@@ -29,31 +29,57 @@ namespace journal {
 //                    not_null<Player::PointerMap*> const pointer_map);
 //  };
 
+template<typename T>
```

Comment:
Put this in an `internal` namespace to make it clear that this is not part of the API.